### PR TITLE
PEP8 fixes in dvsim

### PR DESCRIPTION
**Pull request by rswarbrick**
The first patch corrects a minor bug in Modes.py (presumably the relevant code never ran?). I've split it out because I'm not completely certain the change is correct. @sriyerg ?

The second patch corrects all the other PEP8 errors that flake8 reports in the codebase. The changes are mostly pretty mechanical; the only interesting one is probably the change to `merg_dicts`.

Here's the commit message from that patch:

> [dvsim] PEP8 fixes in dvsim
> 
> These should cause no functional change. Detailed list of code changes:
> 
>   - Get rid of 'import *': this defeats static analysis tools. Don't
>     do it.
> 
>   - Add newline after colon in "if foo: bar"
> 
>   - Use 'is' to check for equality with boolean literals
> 
>   - Don't catch exceptions when running os.system in Deploy.py: the
>     os.system function returns the program's exit code.
> 
>   - Delete some variables that are written but not read.
> 
>   - Minor whitespace changes (missing blank lines between functions;
>     weird indentation; missing space after '#')
> 
>   - Delete autogenerated module docstrings (they didn't contain any
>     information. Maybe it would be good to have a docstring, but at
>     the moment it's just noise).
> 
>   - Don't use \ as a line continuation character. Use parentheses if
>     necessary.
> 
>   - Replace code like "foo" + \ "bar" with just "foo" "bar" (Python
>     concatenates adjacent string literals just like C). (I didn't do
>     this everywhere, but it happened a lot next to the backslash
>     continuations, so I got rid of the unnecessary '+' then).
> 
>   - Replace "not foo in bar" with "foo not in bar"
> 
>   - Use raw strings for regexes with backslashes (r'a\+', not 'a\+')
> 
>   - General cleanup in utils.py's merge_dicts function. This caused
>     flake8 errors because it compared equality of types. The new code
>     avoids the errors and I think it's generally a bit easier to
>     understand.

### CODE REVIEW

**Comment by rswarbrick** [comment]
In case anyone is wondering why I've been hacking on this: I was trying to understand how `dvsim.py` deals with seeds and repetition counts etc. and had to trace through the code to figure out what was going on. The `from blah import *` lines made this rather harder, so I decided to just fix it up for next time.

**Comment by rswarbrick** [comment]
Re-push was because the first CI run failed to download Hugo and errored out. Same problem the second time. Boo! So please could someone have a look at the code? I suspect that there will be some review comments to address, giving CI ample opportunity to work in the future :-)

File: util/dvsim/testplanner/testplan_utils.py [hunk 1]
```diff
@@ -132,32 +132,43 @@ def parse_hjson(filename):
...
-            else:
-                print("The type of value of key \"", key, "\" in list1: \"", \
-                      str(type(list1[key])), \
-                      "\" does not match the type of value in list2: \"", \
-                      str(type(list2[key])), \
-                      "\". The two lists cannot be merged.")
-                sys.exit(1)
-        else:
-            list1[key] = list2[key]
+    for key, item2 in list2.items():
```
Comment (sriyerg) [inline util/dvsim/testplanner/testplan_utils.py:143]: Thanks for rewriting it, LG! Would you please move this change to a separate commit though, since it is not really a PEP8 fix?

**Comment by sriyerg** [comment]
I also added other people who contributed to this tool to review the changes.

File: util/dvsim/Modes.py [hunk 2]
```diff
@@ -509,7 +509,7 @@ def create_regressions(regdicts, sim_cfg, tests):
                 # which case, skip
                 if run_mode_obj.name in sim_cfg.en_run_modes:
                     continue
-                self.run_opts.extend(run_mode_obj.run_opts)
+                regression_obj.run_opts.extend(run_mode_obj.run_opts)
```
[the remark this answers is not in the document]
Comment (rswarbrick) [reply util/dvsim/Modes.py:516]: Ok, done (issue #2056). I don't know the steps to reproduce the bug, I'm afraid. I *think* this is the right fix, but to be certain, either the author of the code (you?) or a user (who can't yet exist, because it doesn't work) would need to check it.

File: util/dvsim/testplanner/testplan_utils.py (hunk 1 above)
Comment (rswarbrick) [reply util/dvsim/testplanner/testplan_utils.py:143]: Well, I only looked at the code because the type comparison triggers a PEP8 error. But I'll split it out into a separate commit if you think that would be better: I'll push up a new version in a few minutes. (Hopefully that will trigger a new CI run, which might work now).

**Comment by rswarbrick** [comment]
New version splits out the `merge_dicts` change into a separate commit as requested and rebases onto current master.

File: util/dvsim/Deploy.py [hunk 3]
```diff
@@ -276,7 +278,8 @@ def set_status(self):
                 self.status = "F"
 
             # Return if status is fail - no need to look for pass patterns.
-            if self.status == 'F': return
+            if self.status == 'F':
```
[the remark this answers is not in the document]
Comment (sriyerg) [reply util/dvsim/Deploy.py:281]: Ohh, I am just interested in knowing if there is an easy way to deal with mistakes like this. yapf allows it.

File: util/dvsim/FlowCfg.py [hunk 4]
```diff
@@ -1,19 +1,19 @@
...
 from shutil import which
+import subprocess
+import sys
 
 import hjson
 
-from Deploy import *
-from utils import *
+from Deploy import Deploy
+from utils import VERBOSE, md_results_to_html, parse_hjson, subst_wildcards
```
[the remark this answers is not in the document]
Comment (sriyerg) [reply util/dvsim/FlowCfg.py:16]: Thanks for the suggestion, will definitely do it!

File: util/dvsim/Modes.py [hunk 5]
```diff
@@ -134,7 +132,8 @@ def create_modes(ModeType, mdicts):
         def merge_sub_modes(mode, parent, objs):
             # Check if there are modes available to merge
             sub_modes = mode.get_sub_modes()
-            if sub_modes == []: return
+            if sub_modes == []:
```
[the remark this answers is not in the document]
Comment (sriyerg) [reply util/dvsim/Modes.py:135]: Yes, that's fine. I am not a Python expert by any means. It seems like you are, so this question is more for my own understanding :)

File: util/dvsim/testplanner/testplan_utils.py (hunk 1 above)
Comment (sriyerg) [reply util/dvsim/testplanner/testplan_utils.py:143]: This looks functionally equivalent to the old code, but we cannot say for sure unless we test it. Moving it to a separate commit allows an easier path to roll back if needed. Thanks for splitting it!

File: util/dvsim/Deploy.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (rswarbrick) [reply util/dvsim/Deploy.py:281]: Understood about yapf. I guess flake8 is just differently opinionated. :-) Reading through the actual text of the PEP8 recommendation, I found this:

> While sometimes it’s okay to put an if/for/while with a small body on the same line, never do > this for multi-clause statements. Also avoid folding such long lines!

So it seems that flake8 is being more grumpy than necessary. That said, I'm probably on flake8's side here.

**Comment by rswarbrick** [comment]
> lgtm, but please fix the CI failure:
> 
> > The commit message must contain a Signed-off-by line
> 
> also, we usually have one commit in a PR. You can use rebase to merge them into one commit.

@weicaiyang: Thanks. I'll push up a version with the missing sign-off line on Monday. Can you have a chat with @sriyerg about squashing commits? He explicitly asked me to split the 2 original commits into 3, which I've done. I don't really mind whether this is 1 commit or 3, but I don't think I can satisfy both of you at once :-D

**Comment by weicaiyang** [comment]
> > lgtm, but please fix the CI failure:
> > > The commit message must contain a Signed-off-by line
> > 
> > 
> > also, we usually have one commit in a PR. You can use rebase to merge them into one commit.
> 
> @weicaiyang: Thanks. I'll push up a version with the missing sign-off line on Monday. Can you have a chat with @sriyerg about squashing commits? He explicitly asked me to split the 2 original commits into 3, which I've done. I don't really mind whether this is 1 commit or 3, but I don't think I can satisfy both of you at once :-D

I think 3 commits are also fine. Merged. Thanks for fixing these.
